hypershift-operator/metrics.go

```diff
@@ -20,11 +23,53 @@ type hypershiftMetrics struct {
 	// repeatedly with the same value.
 	clusterCreationTime *prometheus.GaugeVec
 
+	hostedClusters              *prometheus.GaugeVec
+	hostedClustersWithCondition *prometheus.GaugeVec
+	hostedClustersNodePools     *prometheus.GaugeVec
+	nodePools                   *prometheus.GaugeVec
+	nodePoolsWithCondition      *prometheus.GaugeVec
+	nodePoolSize                *prometheus.GaugeVec
+
 	client crclient.Client
 
 	log logr.Logger
 }
 
+func newMetrics(client crclient.Client, log logr.Logger) *hypershiftMetrics {
+	return &hypershiftMetrics{
+		clusterCreationTime: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Help: "Time in seconds it took from initial cluster creation and rollout of initial version",
+			Name: "hypershift_cluster_initial_rollout_duration_seconds",
+		}, []string{"name"}),
+		hostedClusters: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "hypershift_hostedclusters",
+			Help: "Total number of HostedClusters by platform, endpoint_access, version",
+		}, []string{"platform", "endpoint_access", "version"}),
+		hostedClustersWithCondition: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "hypershift_hostedclusters_conditions",
+			Help: "Total number of HostedClusters by platform with conditions in undesired state",
+		}, []string{"platform", "condition"}),
```

Comment:
Maybe name it `failure_conditions` or sth? The help text is not visible in Prom and that label name is not particularly self-explanatory